### PR TITLE
TRUS-3591 Add Trust Connector and Service calls

### PR DESCRIPTION
**Pull request by carrielies**

### CODE REVIEW

File: app/controllers/asset/AddAssetsController.scala [hunk 1]
```diff
@@ -66,35 +66,39 @@ class AddAssetsController @Inject()(
...
 
-      val assets = new AddAssetViewHelper(userAnswers).rows
-
-      val maxLimit: Int = (userAnswers.is5mldEnabled, isTaxable) match {
-        case (true, true) => MAX_5MLD_TAXABLE_ASSETS
-        case (true, false) => MAX_5MLD_NON_TAXABLE_ASSETS
-        case _ => MAX_4MLD_ASSETS
-      }
+      for {
+        assets <- trustService.getAssets(userAnswers.identifier)
```
Comment (jackheslop96) [inline app/controllers/asset/AddAssetsController.scala:76]: Don't forget to cleanup user answers here. Happy to merge without though as work in progress

File: app/models/Assets.scala [hunk 2]
```diff
@@ -18,22 +18,33 @@ package models
...
-  implicit val assetsFormat: Format[Assets] = Json.format[Assets]
+  implicit val reads: Reads[Assets] =
+    ((__ \ "assets" \ "monetary").readWithDefault[List[AssetMonetaryAmount]](Nil)
+      and (__ \ "assets" \ "propertyOrLand").readWithDefault[List[PropertyLandType]](Nil)
+      and (__ \ "assets" \ "shares").readWithDefault[List[SharesType]](Nil)
+      and (__ \ "assets" \ "business").readWithDefault[List[BusinessAssetType]](Nil)
+      and (__ \ "assets" \ "partnerShip").readWithDefault[List[PartnershipType]](Nil)
+      and (__ \ "assets" \ "other").readWithDefault[List[OtherAssetType]](Nil)
+      and (__ \ "assets" \ "nonEEABusiness").readWithDefault[List[NonEeaBusinessType]](Nil)
+      ).apply(Assets.apply _)
```
Comment (jackheslop96) [inline app/models/Assets.scala:44]: I'm wondering if these reads need the `\ "assets" \`? I presume they're applied when getting the assets from the backend but unless I'm mistaken that picks the data from the json at the `assets` level

File: app/utils/AddAssetViewHelper.scala [hunk 3]
```diff
@@ -17,136 +17,88 @@
...
-      name = nebvm.label.getOrElse(defaultName),
-      typeLabel = nebvm.`type`.label,
-      changeUrl = if (nebvm.status == Completed) {
-        noneeabusiness.routes.AnswersController.onPageLoad().url
-      } else {
-        noneeabusiness.routes.NameController.onPageLoad(NormalMode).url
-      },
+      name = asset.orgName,
+      typeLabel = messages(s"entities.asset.nonEeaBusiness"),
+      changeUrl = noneeabusiness.routes.NameController.onPageLoad(NormalMode).url,
```
Comment (jackheslop96) [inline app/utils/AddAssetViewHelper.scala:101]: Should redirect to answers page

File: app/models/Assets.scala (hunk 2 above)
Comment (carrielies) [reply app/models/Assets.scala:44]: They are needed. I had to add the assets part after doing a manual test and finding that none of the data from trusts was being picked up

File: app/controllers/asset/AddAssetsController.scala (hunk 1 above)
Comment (carrielies) [reply app/controllers/asset/AddAssetsController.scala:76]: Yep. This is my next task, completely updating all this code and re-writing all the tests

File: app/utils/AddAssetViewHelper.scala (hunk 3 above)
Comment (carrielies) [reply app/utils/AddAssetViewHelper.scala:101]: Ok, I didn't know that. I'll fix that

Comment (carrielies) [reply app/utils/AddAssetViewHelper.scala:101]: Actually, Richy will as the Amend is his ticket
